stable/traefik/templates/deployment.yaml

```diff
@@ -59,6 +59,10 @@ spec:
           periodSeconds: 10
           successThreshold: 1
           timeoutSeconds: 2
+        {{- if .Values.env }}
+        env:
+{{ toYaml .Values.env | indent 8 }}
```

Comment:
This has me a little concerned. I know this is the easy way to shim a whole bunch of environment variables (which will vary from one DNS provider to the next) into the pods, _but_... we're mainly talking about username/passwords or secret keys, etc. here. i.e. All things that really belong in a secret. So, while this works, I think there's some less-than-ideal exposure of credentials here.

Reply:
And those secrets would be managed separately from the chart? tbh, that doesn't seem like an ideal user experience.

You're enabling a great degree of flexibility here by allowing freeform entry of env variables and values (or values from secrets), but it seems like lots of rope to hang oneself with, no?

I know from a contribution perspective, it probably seems less appealing to build in specific support for each DNS provider, but imho, that's the way to go if you want a maximally optimal UX for users of this feature.

If the above is too daunting a proposition, maybe it's worth it to add support for a few of the most common DNS providers first and others can be contributed over time?

Reply:
@krancour 

> I know from a contribution perspective, it probably seems less appealing to build in specific support for each DNS provider, but imho, that's the way to go if you want a maximally optimal UX for users of this feature.

That makes a lot of sense. I'll work on that. Thanks!

Reply:
@krancour Want to run something by you before I implement it. The `dnsProvider` configuration will look like this:
```yaml
acme:
  enabled: true
  ...
  dnsProvider:
    route53:
      AWS_REGION: us-east-1
      AWS_ACCESS_KEY_ID: ...
      AWS_SECRET_ACCESS_KEY: ...
```

The `AWS_*` keys will be created as a kubernetes secret and injected into the pod via the deployment using env vars.

Does that sound like a good user interface for this feature?

Reply:
I think you need a field still that lets the user specify _which_ of the many possible DNS providers to use, but overall, yes. This is very much what I had in mind. Whether now or in the future, more sections can be added for each provider. Both the secret and the deployment will have big `if / else if / else if / ... / else if` blocks to include the right fields in the secret and then for the deployment to grab the right env vars from the right fields in the secret.

What I like about this approach, as opposed to the map of environment variables, is that its more constrained. A glance at `values.yaml` (or `helm inspect traefik values`) is all you need to comprehend what options are available to you and how to set them. It's one-stop shopping instead of sending users out-of-band to peruse the Traefik docs.

👍